nanodbc/nanodbc.cpp

```diff
@@ -5167,6 +5167,11 @@ NANODBC_INSTANTIATE_TVP_BINDS(timestamp);
 NANODBC_INSTANTIATE_TVP_BIND_STRINGS(std::string);
 NANODBC_INSTANTIATE_TVP_BIND_STRINGS(wide_string);
 
+#ifdef NANODBC_SUPPORT_STRING_VIEW
```

Comment:
How about replacing this with the C++ [feature test](https://en.cppreference.com/w/cpp/utility/feature_test):
```cpp
#if __cpp_lib_string_view >= 201803L
```
or something similar w/o custom define, as I did in https://github.com/nanodbc/nanodbc/pull/148

Reply:
hmm. It should be separated PR (NANODBC_SUPPORT_STRING_VIEW defined other position, and it is already in use)
imho, we need configuration header, for prebuilt libraries. 
cmake feature options and some c++ feature test options should be there.

Reply:
> imho, we need configuration header, for prebuilt libraries.

I'm not clear why we need it.
We have very few `#ifdef`-s and I'd prefer to keep it simple w/o generating anything, but using pre-defined macros, e.g.:

```cpp
#if ((defined(_MSVC_LANG) && _MSVC_LANG >= 201703L) || __cplusplus >= 201703L)
... // string_view is available
#endif
```

Reply:
1. Build library using C++14
2. If someone take prebuilt nanodbc library and build executable with C++17, header/source compiler version mismatch occured. maybe they found symbol not found error
3. Library consumers should check prebuilt library options and C++ versions, and should set each programs.

Reply:
You mean packaging perspective. Well, that's something we could take into account. /cc @lexicalcast

Reply:
I suppose if it's documented in the README then then it could be ok. Maybe you don't want to use `std::string_view` even if your compiler supports it? 🤷🏻‍♀️

Reply:
@lexicalunit The issue is, AFAIU, that if we offers toggles to freely configure what features are built-in, then it makes life of packagers difficult and the build should also generate public header that indicates what feature is built in.